src/ripple/rpc/handlers/WalletPropose.cpp

```diff
@@ -82,22 +82,13 @@ Json::Value walletPropose (Json::Value const& params)
 
         if (keyType == KeyType::invalid)
             return rpcError(rpcINVALID_PARAMS);
-
-        if (params.isMember (jss::passphrase) ||
-            params.isMember (jss::seed) ||
-            params.isMember (jss::seed_hex))
-        {
-            seed = RPC::getSeedFromRPC (params);
-        }
-        else
-        {
-            seed = randomSeed ();
-        }
     }
-    else if (params.isMember (jss::passphrase))
+
+    if (params.isMember (jss::passphrase) ||
+        params.isMember (jss::seed) ||
+        params.isMember (jss::seed_hex))
     {
-        seed = parseGenericSeed (
-            params[jss::passphrase].asString());
+        seed = RPC::getSeedFromRPC (params);
```

Comment:
Lines 98 and 99 have:

    if (!seed)
        return rpcError(rpcBAD_SEED);

Please move them here, inside this `if`.